### DISCUSSION TITLE
Trouble navigating documentation

### DISCUSSION THREAD

**Discussion by SelfMade2001** [bcakmakoglu/vue-flow#1487] [Q&A] [answered]
This might be on me, but I feel like I do not understand the way this package is documented. I really like the [guide](https://vueflow.dev/guide/) from Introduction to [State](https://vueflow.dev/guide/vue-flow/state.html) but the following three sections ([actions](https://vueflow.dev/typedocs/interfaces/Actions.html), [getters](https://vueflow.dev/typedocs/interfaces/Getters.html) and [events](https://vueflow.dev/typedocs/interfaces/FlowEvents.html)) just link to the TypeDocs without explanation. I get that everything in Typescript needs a type even though I am only using javascript without Typescript. But I do not get which of the stuff from the TypeDocs I can use and where I can use it.

I just feel like I am missing something, because I keep discovering events that are emitted only through the [examples](https://vueflow.dev/examples/) section. Or I stumble over discussions like [this one](https://github.com/bcakmakoglu/vue-flow/discussions/1429) and apparently there is something called `ConnectionMode.Loose`, but I do not know where I can set this mode. So I search the docs with the search feature but can only find it as a Typescript type.

Moreover, I get that the useVueFlow composable is important, but I do not get which stuff I can pull from it. Apparently I can use it in order to get onInit, findNode, fitView, snapToGrid [(found here)](https://vueflow.dev/guide/composables.html#usevueflow) but also much more. How can I find out the complete list of stuff I can pull from useVueFlow?

**Comment by bcakmakoglu** [comment, marked as answer]
> But I do not get which of the stuff from the TypeDocs I can use and where I can use it.

Depends on what you're looking for? It's literally just turning TypeScript definitions into a documentation and allowing users such as yourself that don't even use TypeScript to still look up, for example, what arguments a function would take and what these arguments should be (like [`ValidConnectionFunc`](https://vueflow.dev/typedocs/types/ValidConnectionFunc.html))

> I just feel like I am missing something

Well, be specific. *What* are you missing?

> because I keep discovering events that are emitted only through the [examples](https://vueflow.dev/examples/) section.

I can add some docs about that as well, although the examples showcase that functionality already and my time is limited 😐

>  but I do not know where I can set this mode

Try `<VueFlow :connection-mode="ConnectionMode.Loose">` - [this option is documented](https://vueflow.dev/guide/vue-flow/config.html#connection-mode-optional) (not to a crazy extent though ^^)

> Moreover, I get that the useVueFlow composable is important, but I do not get which stuff I can pull from it.

*What* are you looking for?

> How can I find out the complete list of stuff I can pull from useVueFlow?

[In the TypeDocs of `useVueFlow`.](https://vueflow.dev/typedocs/functions/useVueFlow.html)
(green texts are usually links that you can click.)

I do plan to extend the docs more in the future (probably together with v2).
Thank you for the post, the feedback is highly appreciated as it gives me an idea on where the docs need to be clearer (although the `actions`, `getters` and `events` links are just there so that users have *something* to reference instead of nothing at all 😅).

(Tip: Most of what applies to [ReactFlow](https://reactflow.dev/) also applies to VueFlow. Their docs are a lot more extensive and can be very helpful.)

**Comment by SelfMade2001** [comment]
Thank you, I really appreciate your quick reply! :D

I do understand useVueFlow better now. The confusion came from not knowing what the VueFlowStore should be excatly and being overwhelmed by the answer presented [here](https://vueflow.dev/typedocs/types/VueFlowStore.html). (I am not used to Typescript type definitions.)

Here is what I think the VueFlowStore is: It contains every [event](https://vueflow.dev/typedocs/interfaces/FlowEvents.html) VueFlow offers + everything in the [state](https://vueflow.dev/typedocs/interfaces/State.html) + every [getter](https://vueflow.dev/typedocs/interfaces/Getters.html) in VueFlow + every [action](https://vueflow.dev/typedocs/interfaces/Actions.html) in VueFlow.

This is consistent with looking in the console after calling `console.log(useVueFlow());`.

**Reply by bcakmakoglu** [reply]
> it contains every [event](https://vueflow.dev/typedocs/interfaces/FlowEvents.html) VueFlow offers + everything in the [state](https://vueflow.dev/typedocs/interfaces/State.html) + every [getter](https://vueflow.dev/typedocs/interfaces/Getters.html) in VueFlow + every [action](https://vueflow.dev/typedocs/interfaces/Actions.html) in VueFlow. 

Yep, `useVueFlow` exposes basically everything ^^
